query/src/sql/sql_parser.rs

```diff
@@ -162,363 +154,24 @@ impl<'a> DfParser<'a> {
-            // bigdecimal feature is enabled, and is otherwise a no-op
-            // (i.e., it returns the input string).
-            Token::Number(ref n, l) => match n.parse() {
-                Ok(n) => Ok(Value::Number(n, l)),
-                Err(e) => parser_err!(format!("Could not parse '{}' as number: {}", n, e)),
-            },
-            Token::SingleQuotedString(ref s) => Ok(Value::SingleQuotedString(s.to_string())),
-            Token::NationalStringLiteral(ref s) => Ok(Value::NationalStringLiteral(s.to_string())),
-            Token::HexStringLiteral(ref s) => Ok(Value::HexStringLiteral(s.to_string())),
-            unexpected => self.expected("a value", unexpected),
-        }
-    }
-
-    pub(crate) fn parse_value_or_ident(&mut self) -> Result<String, ParserError> {
-        match self.parser.next_token() {
             Token::Word(w) => match w.keyword {
-                Keyword::TRUE => Ok("true".to_string()),
-                Keyword::FALSE => Ok("false".to_string()),
-                Keyword::NULL => Ok("null".to_string()),
-                _ => Ok(w.value),
-            },
-            // The call to n.parse() returns a bigdecimal when the
-            // bigdecimal feature is enabled, and is otherwise a no-op
-            // (i.e., it returns the input string).
-            Token::Number(n, _) => Ok(n),
-            Token::SingleQuotedString(s) => Ok(s),
-            Token::NationalStringLiteral(s) => Ok(s),
-            Token::HexStringLiteral(s) => Ok(s),
-            unexpected => self.expected("a value", unexpected),
-        }
-    }
-
-    fn parse_alter(&mut self) -> Result<DfStatement<'a>, ParserError> {
-        match self.parser.next_token() {
-            Token::Word(w) => match w.keyword {
-                Keyword::USER => self.parse_alter_user(),
-                Keyword::FUNCTION => self.parse_alter_udf(),
-                Keyword::TABLE => self.parse_alter_table(),
-                Keyword::VIEW => self.parse_alter_view(),
-                Keyword::DATABASE => self.parse_alter_database(),
-                _ => self.expected("keyword USER or FUNCTION", Token::Word(w)),
-            },
-            unexpected => self.expected("alter statement", unexpected),
-        }
-    }
-
-    fn parse_describe(&mut self) -> Result<DfStatement<'a>, ParserError> {
-        match self.parser.next_token() {
-            Token::Word(w) => match w.keyword {
-                Keyword::TABLE => self.parse_desc_table(),
-                Keyword::STAGE => self.skip_all(DfStatement::DescribeStage),
-
-                _ => {
-                    self.parser.prev_token();
-                    self.parse_desc_table()
+                Keyword::EXPLAIN => {
+                    self.parser.next_token();
+                    self.parse_explain()
                 }
+                Keyword::INSERT => self.parse_insert(),
+                Keyword::SELECT | Keyword::WITH | Keyword::VALUES => self.parse_query(),
```

Comment:
parse_query may return error

Reply:
> parse_query may return error

This PR didn't add the fallback logic. I expect it to be added in the next PR.

Reply:
> but it will never return an error anymore.

Oh, yes. I'm wrong about this statement.

The older parser will not return an error except `select`, `insert`, `delete` and `explain`.